js/src/components/tree-select-control/index.js

```diff
@@ -163,52 +271,107 @@ const TreeSelectControl = ( {
 
 		let filteredTreeOptions = cloneDeep( treeOptions );
 
-		if ( filter ) {
-			filteredTreeOptions = filteredTreeOptions.filter( filterOption );
-		}
-
+		filteredTreeOptions = filteredTreeOptions.filter( filterOption );
 		filteredOptionsCache.current[ filter ] = filteredTreeOptions;
 		setFilteredOptions( filteredTreeOptions );
-	}, [ treeOptions, filter ] );
+	}, [ treeOptions, filter, getOption ] );
 
 	const onKeyDown = ( event ) => {
-		if ( inputControlValue ) return;
+		if ( ESCAPE === event.key ) {
+			setTreeVisible( false );
+		}
 
-		if ( BACKSPACE === event.key ) {
-			onChange( value.slice( 0, -1 ) );
+		if ( ENTER === event.key ) {
+			setTreeVisible( true );
 			event.preventDefault();
 		}
 
-		if ( ESCAPE === event.key ) {
-			setTreeVisible( false );
+		if ( inputControlValue ) return;
+
+		if ( ARROW_UP === event.key ) {
+			const parent = getParent( focused.el );
+			const child = parent?.children[ focused.idx - 1 ];
+
+			if ( ! parent ) {
+				const el = root
+					? getLastChild( root )
```

Comment:
🚧  or ❓ 
Same as https://github.com/woocommerce/google-listings-and-ads/pull/1336#discussion_r829717689.